scio-macros/src/main/scala-3/com/spotify/scio/IsJava.scala

```diff
@@ -59,15 +59,18 @@ object IsJavaBean {
             report.throwError(mess)
           }
 
-      val resType = info.returnTpt.tpe
-      val paramType = setter.paramss.head.head.tpt.tpe
-
-      if (resType != paramType) {
-          val mess =
-            s"""JavaBean contained setter for field $name that had a mismatching type.
-                  |  found:    $paramType
-                  |  expected: $resType""".stripMargin
-          report.throwError(mess)
+      val resType: TypeRepr = info.returnTpt.tpe
+      setter.paramss.head match {
+        case TypeParamClause(params: List[TypeDef]) => report.throwError(s"JavaBean setter for field $name has type parameters")
+        case TermParamClause(head :: _) => 
+          val tpe = head.tpt.tpe
+          if (resType != tpe) {
+            val mess =
+              s"""JavaBean contained setter for field $name that had a mismatching type.
+                    |  found:    $tpe
+                    |  expected: $resType""".stripMargin
+            report.throwError(mess)
+          }
```

Comment:
Are the changes here needed for the CI to be green? If not, I'd suggest delay the change to another PR to fix compilation of core.

Reply:
This changes are necessary for the code to compile